### PR TITLE
Fix json parse errors

### PR DESCRIPTION
**Pull request by marcbachmann**
### Changelog
- 🐛 Handle json parse errors of the handshake object

### CODE REVIEW

**Comment by marcbachmann** [comment]
Also, would you like to give me permission to clean up a few of the issues/prs and run the tests? ☺️

File: src/lib.rs [hunk 1]
```diff
@@ -154,14 +154,17 @@ impl OIDCFilter {
         };
 
         let json = serde_json::to_string(&claims)?;
-        let flags = if claims.proto.as_str() == "http" { "HttpOnly" } else { "HttpOnly; Secure" };
+        let flags = if claims.proto.as_str() == "http" { "HttpOnly" } else { "HttpOnly; Secure; SameSite=Lax" };
```
Comment (marcbachmann) [inline src/lib.rs:157]: Hmm, that still didn't work

Comment (marcbachmann) [reply src/lib.rs:157]: oh, default is SameSite=Lax. SameSite=None is rerequired 🤦

File: src/lib.rs [hunk 2]
```diff
@@ -184,7 +187,17 @@ impl HttpContext for OIDCFilter {
 
         let code = self.get_code();
         if code != "" {
-            let handshake = self.get_handshake_object().unwrap();
+            let handshake = match self.get_handshake_object() {
+                Ok(handshake) => handshake,
+                Err(_e) => {
```
Comment (dgn) [inline src/lib.rs:192]: we should probably log this error

**Comment by dgn** [comment]
This will need a rebase now that https://github.com/dgn/oidc-filter/pull/15 has been merged

**Comment by marcbachmann** [comment]
done

File: src/lib.rs (hunk 2 above)
Comment (marcbachmann) [reply src/lib.rs:218]: I guess that's good enough:
```rust
error!("Failed to parse handshake object: {}", err);
```

**Review by dgn — APPROVED** [review]
Thank you very much @marcbachmann!

**Pull request merged**